tests/libres_tests/res/job_queue/test_job_queue.py

```diff
@@ -264,3 +272,41 @@ def test_add_ensemble_evaluator_info_cert_none(self):
 
                 assert content["ee_cert_path"] == None
                 assert not (runpath / cert_file).exists()
+
+
+@pytest.mark.timeout(20)
+@pytest.mark.asyncio
+async def test_retry_on_closed_connection():
+    with TestAreaContext("job_queue_test_retry"):
+        job_queue = create_queue(SIMPLE_SCRIPT, max_submit=1)
+        pool_sema = BoundedSemaphore(value=10)
+
+        with patch("websockets.connect") as f:
+            websocket_mock = AsyncMock()
+            f.side_effect = [
+                ConnectionClosedError(1006, "expected close"),
+                websocket_mock,
```

Comment:
Would I interpret this correct as the while loop in `_publish_changes` will now never be able to connect, simply exhausting the retry count with the increased backoff? Hence this test will always time out?

Reply:
No, it will get one connection error and one mocked connection which it can use to send stuff…

Reply:
But many more connections are made, which will fail, but the retry logic has been demonstrated before that happens. Let me know if you want me to write this differently.

Reply:
Ah, it's a list..